### PR TITLE
chore: cleanup exports, remove unused hooks

### PR DESCRIPTION
**Pull request by layershifter**
This PR removes some exports from the existing packages to avoid exports of private APIs.

### `@fluentui/react-components`

- `getNativeProps()` - `getNativeElementProps()` should be used instead
- `useSSRContext()` - provides internal value, `useIsSSR()` should be used

### `@fluentui/react-utilities`

- `*Properties` - these constants should be not exported, they are parts of the `getNativeElementProps()` implementation
- `AsIntrinsicElement`, `IsSingleton`, `PropsWithoutRef`, `ReplaceNullWithUndefined`, `UnionToIntersection`, `UnknownSlotProps` - internal types
- `useBoolean()`, `useMount()` & `useUnmount` - never used, exports and hooks are removed
- `UseBooleanCallbacks` - never used
- `useConst()` - it's recommended to use [`useState()`](https://reactjs.org/docs/hooks-faq.html#how-to-create-expensive-objects-lazily) in React docs. No reason to have this hook
- `defaultSSRContextValue` - internal constant, should not be exported
- `SSRContext` - should not be exported, fix ESLint exception
- `getNativeProps()` - `getNativeElementProps()` should be used instead
- `isFluentTrigger()` - should not be exported, used internally

### CODE REVIEW

File: packages/react-components/react-utilities/src/utils/properties.ts [hunk 1]
```diff
@@ -421,7 +416,8 @@ export const divProperties = htmlElementProperties;
  *
  * @public
  * @param props - The unfiltered input props
- * @param allowedPropsNames - The array or record of allowed prop names.
+ * @param allowedPropNames - The array or record of allowed prop names.
+ * @param excludedPropNames
```
Comment (layershifter) [inline packages/react-components/react-utilities/src/utils/properties.ts:420]: Updated JSDoc to match implementation

File: packages/react-components/react-utilities/src/utils/properties.ts [hunk 2]
```diff
@@ -398,11 +398,6 @@ export const imgProperties = toObjectMap(htmlElementProperties, [
   'width', // canvas, embed, iframe, img, input, object, video
 ]);
 
-/**
- * @deprecated Use imgProperties for img elements.
- */
-export const imageProperties = imgProperties;
-
```
Comment (layershifter) [inline packages/react-components/react-utilities/src/utils/properties.ts:405]: It was deprecated 👀

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                                                                             |      Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                                    Change |
| :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: | ----------------------------: | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-accordion</samp> <br /> <abbr title='bundle-size/Accordion.fixture.js'>Accordion (including children components)</abbr>                                           |  `74.243 kB`<br />`22.679 kB` |  `74.036 kB`<br />`22.588 kB` | `-207 B` <br />`-91 B`  |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr>                                                                                    |  `45.571 kB`<br />`13.318 kB` |  `45.454 kB`<br />`13.251 kB` | `-117 B` <br />`-67 B`  |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/Badge.fixture.js'>Badge</abbr>                                                                                       |    `20.985 kB`<br />`6.62 kB` |   `20.929 kB`<br />`6.556 kB` |  `-56 B` <br />`-64 B`  |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/CounterBadge.fixture.js'>CounterBadge</abbr>                                                                         |   `21.898 kB`<br />`6.925 kB` |   `21.833 kB`<br />`6.861 kB` |  `-65 B` <br />`-64 B`  |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/PresenseBadge.fixture.js'>PresenceBadge</abbr>                                                                       |    `22.21 kB`<br />`6.672 kB` |   `22.126 kB`<br />`6.623 kB` |  `-84 B` <br />`-49 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr>                                                                                    |   `32.919 kB`<br />`9.027 kB` |   `32.846 kB`<br />`8.967 kB` |  `-73 B` <br />`-60 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr>                                                                    |  `39.846 kB`<br />`10.254 kB` |  `39.753 kB`<br />`10.198 kB` |  `-93 B` <br />`-56 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr>                                                                            |   `35.102 kB`<br />`9.777 kB` |   `35.018 kB`<br />`9.724 kB` |  `-84 B` <br />`-53 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr>                                                                          |  `42.353 kB`<br />`11.078 kB` |  `42.259 kB`<br />`11.018 kB` |  `-94 B` <br />`-60 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr>                                                                        |  `47.504 kB`<br />`10.596 kB` |  `47.329 kB`<br />`10.513 kB` | `-175 B` <br />`-83 B`  |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.All.fixture.js'>Card - All</abbr>                                                                                |   `60.746 kB`<br />`17.41 kB` |  `60.641 kB`<br />`17.339 kB` | `-105 B` <br />`-71 B`  |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.fixture.js'>Card</abbr>                                                                                          |  `56.031 kB`<br />`16.147 kB` |  `55.926 kB`<br />`16.083 kB` | `-105 B` <br />`-64 B`  |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardFooter.fixture.js'>CardFooter</abbr>                                                                              |    `7.793 kB`<br />`3.327 kB` |    `7.737 kB`<br />`3.266 kB` |  `-56 B` <br />`-61 B`  |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardHeader.fixture.js'>CardHeader</abbr>                                                                              |    `9.363 kB`<br />`3.841 kB` |    `9.302 kB`<br />`3.776 kB` |  `-61 B` <br />`-65 B`  |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardPreview.fixture.js'>CardPreview</abbr>                                                                            |     `7.765 kB`<br />`3.35 kB` |    `7.709 kB`<br />`3.287 kB` |  `-56 B` <br />`-63 B`  |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox</abbr>                                                                              |  `61.412 kB`<br />`20.982 kB` |  `61.243 kB`<br />`20.906 kB` | `-169 B` <br />`-76 B`  |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `181.147 kB`<br />`50.599 kB` | `180.959 kB`<br />`50.525 kB` | `-188 B` <br />`-74 B`  |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `34.078 kB`<br />`11.098 kB` |   `33.88 kB`<br />`11.017 kB` | `-198 B` <br />`-81 B`  |
| <samp>react-divider</samp> <br /> <abbr title='bundle-size/Divider.fixture.js'>Divider</abbr>                                                                                 |    `15.55 kB`<br />`5.595 kB` |   `15.485 kB`<br />`5.534 kB` |  `-65 B` <br />`-61 B`  |
| <samp>react-image</samp> <br /> <abbr title='bundle-size/Image.fixture.js'>Image</abbr>                                                                                       |   `10.012 kB`<br />`3.982 kB` |    `9.956 kB`<br />`3.923 kB` |  `-56 B` <br />`-59 B`  |
| <samp>react-input</samp> <br /> <abbr title='bundle-size/Input.fixture.js'>Input</abbr>                                                                                       |   `22.163 kB`<br />`7.295 kB` |   `22.009 kB`<br />`7.196 kB` | `-154 B` <br />`-99 B`  |
| <samp>react-label</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Label</abbr>                                                                                       |     `8.57 kB`<br />`3.586 kB` |    `8.514 kB`<br />`3.528 kB` |  `-56 B` <br />`-58 B`  |
| <samp>react-link</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Link</abbr>                                                                                         |    `11.414 kB`<br />`4.64 kB` |    `11.358 kB`<br />`4.58 kB` |  `-56 B` <br />`-60 B`  |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.fixture.js'>Menu (including children components)</abbr>                                                          |  `111.626 kB`<br />`34.05 kB` | `111.457 kB`<br />`33.972 kB` | `-169 B` <br />`-78 B`  |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.Selectable.fixture.js'>Menu (including selectable components)</abbr>                                             | `114.801 kB`<br />`34.521 kB` |  `114.632 kB`<br />`34.44 kB` | `-169 B` <br />`-81 B`  |
| <samp>react-popover</samp> <br /> <abbr title='bundle-size/Popover.fixture.js'>Popover</abbr>                                                                                 | `103.759 kB`<br />`31.549 kB` |  `103.59 kB`<br />`31.473 kB` | `-169 B` <br />`-76 B`  |
| <samp>react-provider</samp> <br /> <abbr title='bundle-size/FluentProvider.fixture.js'>FluentProvider</abbr>                                                                  |   `14.127 kB`<br />`5.299 kB` |   `13.929 kB`<br />`5.216 kB` | `-198 B` <br />`-83 B`  |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/Radio.fixture.js'>Radio</abbr>                                                                                       |  `29.499 kB`<br />`10.102 kB` |  `29.397 kB`<br />`10.043 kB` | `-102 B` <br />`-59 B`  |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroup.fixture.js'>RadioGroup</abbr>                                                                             |   `13.664 kB`<br />`5.486 kB` |    `13.595 kB`<br />`5.42 kB` |  `-69 B` <br />`-66 B`  |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/Select.fixture.js'>Select</abbr>                                                                                    |   `18.321 kB`<br />`6.573 kB` |   `18.252 kB`<br />`6.511 kB` |  `-69 B` <br />`-62 B`  |
| <samp>react-slider</samp> <br /> <abbr title='bundle-size/Slider.fixture.js'>Slider</abbr>                                                                                    |   `25.326 kB`<br />`8.198 kB` |   `25.156 kB`<br />`8.113 kB` | `-170 B` <br />`-85 B`  |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButton.fixture.js'>SpinButton</abbr>                                                                        |  `42.132 kB`<br />`11.895 kB` |  `41.967 kB`<br />`11.814 kB` | `-165 B` <br />`-81 B`  |
| <samp>react-spinner</samp> <br /> <abbr title='bundle-size/Spinner.fixture.js'>Spinner</abbr>                                                                                 |    `18.23 kB`<br />`6.047 kB` |    `18.15 kB`<br />`5.991 kB` |  `-80 B` <br />`-56 B`  |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/Switch.fixture.js'>Switch</abbr>                                                                                    |   `25.455 kB`<br />`8.255 kB` |    `25.37 kB`<br />`8.201 kB` |  `-85 B` <br />`-54 B`  |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Default.fixture.js'>Text - Default</abbr>                                                                        |   `10.904 kB`<br />`4.298 kB` |    `10.848 kB`<br />`4.24 kB` |  `-56 B` <br />`-58 B`  |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Wrappers.fixture.js'>Text - Wrappers</abbr>                                                                      |   `14.226 kB`<br />`4.714 kB` |   `14.158 kB`<br />`4.658 kB` |  `-68 B` <br />`-56 B`  |
| <samp>react-textarea</samp> <br /> <abbr title='bundle-size/Textarea.fixture.js'>Textarea</abbr>                                                                              |   `21.207 kB`<br />`7.165 kB` |   `21.057 kB`<br />`7.082 kB` | `-150 B` <br />`-83 B`  |
| <samp>react-tooltip</samp> <br /> <abbr title='bundle-size/Tooltip.fixture.js'>Tooltip</abbr>                                                                                 |  `43.584 kB`<br />`14.992 kB` |  `43.503 kB`<br />`14.968 kB` |  `-81 B` <br />`-24 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                          |        Size (minified/GZIP) |
| ---------------------------------------------------------------------------------------------------------- | --------------------------: |
| <samp>react-overflow</samp> <br /> <abbr title='bundle-size/hooks.fixture.js'>hooks only</abbr>            | `10.792 kB`<br />`4.124 kB` |
| <samp>react-portal</samp> <br /> <abbr title='bundle-size/Portal.fixture.js'>Portal</abbr>                 |   `6.272 kB`<br />`2.17 kB` |
| <samp>react-positioning</samp> <br /> <abbr title='bundle-size/usePopper.fixture.js'>usePositioning</abbr> | `23.828 kB`<br />`8.283 kB` |
| <samp>react-utilities</samp> <br /> <abbr title='bundle-size/SSRProvider.fixture.js'>SSRProvider</abbr>    |        `189 B`<br />`161 B` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/27b2d06931604dcc4e677e56e5d3e0eeb98806e3'>27b2d06931604dcc4e677e56e5d3e0eeb98806e3</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23319/perf-test-react-components/Avatar-mount_base.html">890</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23319/perf-test-react-components/Avatar-mount.html">936</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23319/perf-test-react-components/Button-mount_base.html">561</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23319/perf-test-react-components/Button-mount.html">561</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23319/perf-test-react-components/FluentProvider-mount_base.html">1829</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23319/perf-test-react-components/FluentProvider-mount.html">1933</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23319/perf-test-react-components/FluentProviderWithTheme-mount_base.html">261</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23319/perf-test-react-components/FluentProviderWithTheme-mount.html">276</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23319/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">237</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23319/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">247</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23319/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">332</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23319/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">310</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23319/perf-test-react-components/MakeStyles-mount_base.html">1611</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23319/perf-test-react-components/MakeStyles-mount.html">1598</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by ling1726 — APPROVED** [review]

**Review by bsunderhus — APPROVED** [review]

**Pull request merged**